tests/css_quotes/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,279 @@
+    content: '"';
+
+    /* Single single quote. */
+    content: "'";
+    content: '\\'';
+
+    /* One of each. */
+    content: "\\"'";
+    content: '"\\'';
+
+    /* One of each with unnecessary escapes. */
+    content: "\\"\\'";
+    content: '\\"\\'';
+
+    /* More double quotes than single quotes. */
+    content: "\\"'\\"";
+    content: '"\\'"';
+
+    /* More single quotes than double quotes. */
+    content: "\\"''";
+    content: '"\\'\\'';
+
+    /* Two of each. */
+    content: "\\"\\"''";
+    content: '""\\'\\'';
+
+    /* Single backslash. */
+    content: '\\\\';
+    content: "\\\\";
+
+    /* Backslases. */
+    content: "\\"\\\\\\"\\\\\\\\\\" '\\'\\\\'\\\\\\'\\\\\\\\'";
+    content: '\\'\\\\\\'\\\\\\\\\\' "\\"\\\\"\\\\\\"\\\\\\\\"';
+
+    /* Somewhat more real-word example. */
+    content: "He's sayin': \\"How's it goin'?\\" Don't ask me why.";
+    content: 'He\\'s sayin\\': "How\\'s it goin\\'?" Don\\'t ask me why.';
+
+    /* Somewhat more real-word example 2. */
+    content: "var backslash = \\"\\\\\\", doubleQuote = '\\"';";
+    content: 'var backslash = "\\\\", doubleQuote = \\'"\\';';
+
+    /* Leave all "escapes" alone. */
+    content: "\\Abc4 foo \\n" /* "comment" */ "\\end";
+    content: '\\Abc4 foo \\n' /* 'comment' */ '\\end';
+  }
+}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+@supports (content: one 'two' three 'four') {
+  a[href='foo' y],
+  abbr[title^="It's a trap!"],
+  img[src=''] {
+    /* Simple strings. */
+    content: 'abc';
+    content: 'abc';
+
+    /* Escape. */
+    content: '\\A';
+
+    /* Emoji. */
+    content: '🐶';
+
+    /* Empty string. */
+    content: '';
+    content: '';
+
+    /* Single double quote. */
+    content: '"';
+    content: '"';
+
+    /* Single single quote. */
+    content: "'";
+    content: "'";
+
+    /* One of each. */
+    content: '"\\'';
+    content: '"\\'';
+
+    /* One of each with unnecessary escapes. */
+    content: '"\\'';
+    content: '"\\'';
+
+    /* More double quotes than single quotes. */
+    content: '"\\'"';
+    content: '"\\'"';
+
+    /* More single quotes than double quotes. */
+    content: "\\"''";
+    content: "\\"''";
+
+    /* Two of each. */
+    content: '""\\'\\'';
+    content: '""\\'\\'';
+
+    /* Single backslash. */
+    content: '\\\\';
+    content: '\\\\';
+
+    /* Backslases. */
+    content: "\\"\\\\\\"\\\\\\\\\\" ''\\\\'\\\\'\\\\\\\\'";
+    content: '\\'\\\\\\'\\\\\\\\\\' ""\\\\"\\\\"\\\\\\\\"';
+
+    /* Somewhat more real-word example. */
+    content: "He's sayin': \\"How's it goin'?\\" Don't ask me why.";
+    content: "He's sayin': \\"How's it goin'?\\" Don't ask me why.";
+
+    /* Somewhat more real-word example 2. */
+    content: 'var backslash = "\\\\", doubleQuote = \\'"\\';';
+    content: 'var backslash = "\\\\", doubleQuote = \\'"\\';';
+
+    /* Leave all "escapes" alone. */
+    content: '\\Abc4 foo \\n' '\\end';
+    content: '\\Abc4 foo \\n' '\\end';
```

Comment:
Known issue: https://github.com/prettier/prettier/issues/2127

Would be awesome to fix this one as well. Shouldn't be hard: make a tokenizer that extracts out comments from a css file and add it through prettier comment location placement logic.